### PR TITLE
see #86: add support for reverse proxy headers

### LINKED ISSUE #86: Incorrect WP_HOME and WP_SITEURL values when behind a reverse proxy
### Name and Version

bitnami/wordpress-nginx:6

### What steps will reproduce the bug?

1. Configure a docker-compose.yaml with a wordpress-nginx container and an nginx reverse proxy container
2. Configure nginx to reverse proxy the wordpress-nginx container
3. Try to open the site
4. The site will open but static assets will be broken because WordPress is configured using the local hostname instead of the one configured in the nginx reverse proxy

### What is the expected behavior?

Expected behavior is for the setup script to recognize the proper hostname using the relevant HTTP headers.

### What do you see instead?

A broken WP install.

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by ziodave**
**Description of the change**

Add support for reverse proxy headers in wp-config.php.

**Benefits**

Fixes #86 , allowing successful installation in environments where a reverse proxy is used.

**Possible drawbacks**

None.

**Applicable issues**

#86 

**Additional information**

### CODE REVIEW

**Comment by fmulero** [comment]
Hi @ziodave, thanks a lot for your contribution

I've built your image and something seems wrong, I can not spin up a container from this code:
```
wordpress_1  | wordpress 14:10:57.69 INFO  ==> Installing WordPress
wordpress_1  | PHP Parse error:  syntax error, unexpected identifier "define" in phar:///opt/bitnami/wp-cli/bin/wp-cli.phar/vendor/wp-cli/wp-cli/php/WP_CLI/Runner.php(1277) : eval()'d code on line 170
```

You can reproduce the issue with the docker-compose provided in this repository

**Comment by ziodave** [comment]
> Hi @ziodave, thanks a lot for your contribution
> 
> I've built your image and something seems wrong, I can not spin up a container from this code. This is the error:
> 
> ```
> wordpress_1  | wordpress 14:10:57.69 INFO  ==> Installing WordPress
> wordpress_1  | PHP Parse error:  syntax error, unexpected identifier "define" in phar:///opt/bitnami/wp-cli/bin/wp-cli.phar/vendor/wp-cli/wp-cli/php/WP_CLI/Runner.php(1277) : eval()'d code on line 170
> ```
> 
> You can reproduce the issue with the docker-compose provided in this repository

It's fixed (missing escapes).

**Review by fmulero — CHANGES_REQUESTED** [review]
Great contribution @ziodave, thanks a lot!!

I've checking the issue and possible scenarios and I agree this is the best solution. In my opinion this is a real/common scenario so I think other users sorted it out using rewriting rules, publishing wordpress internally over SSL or any other solution. Keeping in mind that and in order to avoid any collateral effect for those users, I think is worth to enable this feature using an ENV var keeping it disabled by default (to keep the current behaviour by default). What do you think? Could you apply those changes?

**Comment by ziodave** [comment]
> Hi @ziodave, thanks a lot for your contribution
> 
> I've built your image and something seems wrong, I can not spin up a container from this code. This is the error:
> 
> ```
> wordpress_1  | wordpress 14:10:57.69 INFO  ==> Installing WordPress
> wordpress_1  | PHP Parse error:  syntax error, unexpected identifier "define" in phar:///opt/bitnami/wp-cli/bin/wp-cli.phar/vendor/wp-cli/wp-cli/php/WP_CLI/Runner.php(1277) : eval()'d code on line 170
> ```
> 
> You can reproduce the issue with the docker-compose provided in this repository

@fmulero it's implemented as a yes/no flag with name `WORDPRESS_ENABLE_REVERSE_PROXY`.

File: README.md [hunk 1]
```diff
@@ -248,6 +248,7 @@ Available environment variables:
 - `WORDPRESS_MULTISITE_NETWORK_TYPE`: WordPress Multisite network type to enable. Valid values: `subfolder`, `subdirectory`, `subdomain`. Default: **subdomain**
 - `WORDPRESS_MULTISITE_ENABLE_NIP_IO_REDIRECTION`: Whether to enable IP address redirection to nip.io wildcard DNS when enabling WordPress Multisite. This is useful when running on an IP address with subdomain network type. Default: **no**
 - `WORDPRESS_MULTISITE_FILEUPLOAD_MAXK`: Maximum upload file size allowed for WordPress Multisite uploads, in kilobytes. Default: **81920**
+- `WORDPRESS_ENABLE_REVERSE_PROXY`: Enable WordPress support for reverse proxy headers. Default: **no**
```
Comment (fmulero) [inline README.md:251]: Could you move this line to the `User and Site configuration` section?

**Review by fmulero — CHANGES_REQUESTED** [review]
Great! I have only a little suggestion. If you are not able to do it, let us know, the rest looks really good. All tests passed in our side

**Review by fmulero — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (fmulero) [reply README.md:251]: No problem, we could leave it as it is

**Comment by ziodave** [comment]
> Great! I have only a little suggestion. If you are not able to do it, let us know, the rest looks really good. All tests passed in our side

sorry @fmulero I was ooo, just came back today. 🙏

**Pull request merged**